File: Drools_client/solution/Drools.Common/Entities/BaseCustomerFact.cs
using GigaSpaces.Core.Metadata;
using System;

namespace Drools.Common.Entities
{
    [Serializable]
    [SpaceClass(AliasName = "com.c123.demo.real.BaseCustomerFact")]
    public class BaseCustomerFact : BaseFact
    {
        [SpaceProperty(AliasName = "customerId")]
        public int? CustomerID { get; set; }

        public BaseCustomerFact()
        {

        }

        public BaseCustomerFact(BaseAccountOperationFact originFact)
            : base(originFact)
        {
            CustomerID = originFact.CustomerID;
        }


        // To verify that we will not duplicate facts, this is the unique definition of a fact.
        protected override object[] IdComponents
        {
            get
            {
                return new object[]
                {
                    this.NetworkID, this.RequestReference
                };
            }
        }
    }
}

File: xap-drools-integration-master/common/src/main/java/com/c123/demo/real/BaseCustomerFact.java
package com.c123.demo.real;

import java.io.Serializable;

public class BaseCustomerFact extends BaseFact implements Serializable {

	private Integer customerId;

	public BaseCustomerFact() {
		super();
	}  
	
	public Integer getCustomerId() {
		return customerId;
	}

	public void setCustomerId(Integer customerId) {
		this.customerId = customerId;
	}

	@Override
	public String toString() {
		return "BaseCustomerFact [customerId=" + customerId + "]";
	}


	
	
}
